sycl/unittests/Extensions/OneAPIProd.cpp

```diff
@@ -0,0 +1,41 @@
+//==------------ OneAPIProd.cpp --- sycl_ext_oneapi_prod unit tests
+//----------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <sycl/sycl.hpp>
+
+#include <helpers/PiMock.hpp>
+
+#include <gtest/gtest.h>
+
+using namespace sycl;
+
+static bool QueueFlushed = false;
+
+static pi_result redefinedQueueFlush(pi_queue Queue) {
+  QueueFlushed = true;
+  return PI_SUCCESS;
+}
+
+TEST(OneAPIProdTest, PiQueueFlush) {
+  sycl::unittest::PiMock Mock(backend::ext_oneapi_level_zero);
+  sycl::platform Plt = Mock.getPlatform();
+  Mock.redefine<detail::PiApiKind::piQueueFlush>(redefinedQueueFlush);
+  context Ctx{Plt};
+  queue Queue{Ctx, default_selector_v};
+  Queue.ext_oneapi_prod();
+  EXPECT_TRUE(QueueFlushed);
+  sycl::ext::oneapi::experimental::command_graph Graph(Ctx, Queue.get_device());
+  Graph.begin_recording(Queue);
+  try {
+    Queue.ext_oneapi_prod(); // flushing while graph is recording is not allowed
+    EXPECT_TRUE(false);
```

Comment:
I think that it is better to use [`FAIL`](https://google.github.io/googletest/reference/assertions.html#FAIL) directly. You should also be able to do something like:
```c++
FAIL() << "exception is expected when calling `ext_oneapi_prod` while graph is being recorded";
```

You can also use [`EXPECT_THROW`](https://google.github.io/googletest/reference/assertions.html#EXPECT_THROW), but it won't allow you to validate error message or error code associated with the exception